docs/idl/NC-Framework.webidl

```diff
@@ -450,212 +432,218 @@ $macro(BlockDatatypes)
+		attribute	NcString	key				// Manufacturer's unique key to product - model number, SKU, etc
+		attribute	NcString	revisionLevel	// Manufacturer's product revision level code
+		attribute	NcString	brandName		// Brand name under which product is sold
+		attribute	NcString		uuid		// Unique UUID of product (not product instance)
+		attribute	NcString	description		// Text description of product
 	};
 	
-	enum ncResetCause {
-		"powerOn",					// 0 Last reset was caused by device power-on.
-		"internalError",			// 1 Last reset was caused by an internal error.
-		"upgrade",					// 2 Last reset was caused by a software or firmware upgrade.
-		"controllerRequest"			// 3 Last reset was caused by a controller request.
+	enum NcResetCause {
+		"PowerOn",			// 0 Last reset was caused by device power-on.
+		"InternalError",	// 1 Last reset was caused by an internal error.
+		"Upgrade",			// 2 Last reset was caused by a software or firmware upgrade.
+		"ControllerRequest"	// 3 Last reset was caused by a controller request.
 	};
 	
-	enum  ncDeviceGenericState {
-		"normalOperation",			// 0 Device is operating normally.
-		"initializing",				// 1 Device is starting  or restarting.
-		"updating",					// 2 Device is performing a software or firmware update.
+	enum NcDeviceGenericState {
+		"NormalOperation",	// 0 Device is operating normally.
+		"Initializing",		// 1 Device is starting  or restarting.
+		"Updating",			// 2 Device is performing a software or firmware update.
 	};
 
 	interface ncDeviceOperationalState {
-		attribute ncDeviceGenericState generic;
-		attribute ncBlob deviceSpecificDetails; //Device implementation specific details
+		attribute NcDeviceGenericState generic;
+		attribute NcBlob deviceSpecificDetails; //Device implementation specific details
 	};
 $endmacro
 $macro(AgentDatatypes)
 	//  ----------------------------------------------------------------------------------
-	//	A g e n t D a t a t y p e s
+	//	Agent datatypes
 	//  ----------------------------------------------------------------------------------
 	
-	enum ncConnectionStatus {					// Connection status	
-		"Undefined",								// 0	This is the value when there is no receiver
-		"Connected",								// 1	Connected to a stream
-		"Disconnected",								// 2	Not connected to a stream
-		"ConnectionError"							// 3	Connected but broken
+	enum NcConnectionStatus 
+		"Undefined",		// 0 This is the value when there is no receiver
+		"Connected",		// 1 Connected to a stream
+		"Disconnected",		// 2 Not connected to a stream
+		"ConnectionError"	// 3 Connected but broken
 	};
 	
-	interface ncPayloadStatus {				// Received payload status
-		"Undefined",								// 0	This is the value when there's no connection.
-		"PayloadOK",								// 1	Payload type is one we know about and the PDU is well-formed
-		"PayloadFormatUnsupported",					// 2	Payload is not one we know about
-		"PayloadError",								// 3	Some kind of error has occurred
+	interface NcPayloadStatus {
```

Comment:
I think this should be an enum, not an interface.  It appears to be an error in the original file.